### PR TITLE
[DS-4407] Support JDPA debug option (for Eclipse & other IDEs) with DSpace Docker

### PR DESCRIPTION
**Pull request by terrywbrady**
I have had some recent luck debugging Java Tomcat Apps running in Docker with Eclipse.

### The following PR would provide support to for DSpace 7.
https://jira.lyrasis.org/browse/DS-4407

### Launch DSpace Docker with the following command:

`docker-compose -f docker-compose.yml -f dspace/src/main/docker-compose/jdpa-debug.yml up -d`

### Create a remote debug configuration in Eclipse:
image

### Set a breakpoint in the code and invoke that code via the HAL browser:

image

### Eclipse Project Setup

Note: In my Eclipse configuration, I imported the DSpace code into Eclipse "as a Maven Project".  I do not know if this necessary to enable the debugger to locate the necessary source modules.

### CODE REVIEW

**Comment by terrywbrady** [comment]
Per Slack conversation with @kshepherd  and @J4bbi, I wonder if this solution would be applicable to any IDE that supports jdpa debugging.  If so, I recommend renaming eclipse-debug.yml to jdpa-debug.yml.  I would need confirmation from users of other IDE's.

File: dspace/src/main/docker-compose/eclipse-debug.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    - published: 8000
+      target: 8000
+    environment:
+      JPDA_ADDRESS: 8000
+      JPDA_TRANSPORT: dt_socket
+    entrypoint:
+    - /bin/bash
+    - '-c'
+    - |
+      /dspace/bin/dspace database migrate
```
Comment (tdonohue) [inline dspace/src/main/docker-compose/eclipse-debug.yml]: Just an FYI, this simple `database migrate` command is not enough to ensure the database is "ready" before Tomcat starts.  See https://github.com/DSpace/DSpace/pull/2635

**Review by tdonohue — COMMENTED** [review]
@terrywbrady : I'm not against these changes...I haven't tried them out myself though yet.  I do agree that we should likely rename the `eclipse-debug.yml` as this isn't Eclipse specific.

One minor thing to note...I've discovered we have a "race condition" where the database and Tomcat are essentially starting at the same time.  This PR would be affected as well, as you are using similar `entrypoint` settings as currently on `master`.  See PR #2635 for how to fix the race condition.

**Comment by tdonohue** [comment]
Closing as outdated

**Pull request closed without merging**